### ISSUE TITLE
Docs: Clarify purpose of no-useless-template-literal (or rename?)

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#8544]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Reading the docs at face value, I would consider it a bug that the example
```ts
const useless = `I am a useless template literal that could be converted to an ordinary string`;
```
is not flagged by no-useless-template-literal. (If it actually should be flagged, feel free to convert this to a bug!)

The way I understand it though, the rule as-is is really going after useless template _interpolations_. I think this strongly needs clarification in the docs.

Furthermore, I think that the quotes rule (https://typescript-eslint.io/rules/quotes/), which _can_ be used to prohibit useless template literals, and the no-useless-template-literal rule docs ought to contain links to each other to help a user figure out which one to use, depending on what the goal is. (I'm currently setting this up at my day job and getting confused over which one to use 😆)

### Affected URL(s)

https://typescript-eslint.io/rules/no-useless-template-literals/
https://typescript-eslint.io/rules/quotes/

cc @StyleShit

**Comment by kirkwaiblinger** [comment]
Note that the original proposals for this type of rule use the verbage "no-useless-interpolation". 

See 
  - https://github.com/typescript-eslint/typescript-eslint/issues/2846,
  - https://github.com/eslint/eslint/issues/10798

**Comment by JoshuaKGoldberg** [comment]
> no-useless-interpolation

...oh, I think that's a much better name for this rule. D'oh. I wish I'd thought to request the name change back in #7957.

IMO we should rename the rule. Which isn't a process we have documented yet. Filed _(one sec)_.

**Comment by Josh-Cena** [comment]
I don't really like `no-useless-interpolation`, FWIW. `no-useless-template-expression` is better, if you want. `interpolation` is not a uniformly recognized word and has so many meanings—in mathematics it means something entirely different.

**Comment by JoshuaKGoldberg** [comment]
_It has been [0] days since a unified TypeScript glossary would have been useful_

Is there something relative not-overloaded we can call the interpolated insides of a template literal? The `TemplateLiteral` AST node calls it `expressions`... but that term is used in a lot of places. 😕

**Comment by Josh-Cena** [comment]
On MDN, we call it "embedded expression" or just "expression". In the ES spec, it's called "substitution" (in many places). I think both are better than either `interpolation` or `template-literal`.

**Comment by kirkwaiblinger** [comment]
FWIW, the very first sentence on the MDN page uses the term "interpolation" https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals, so I'm skeptical that it's not a widely recognized term.

That said, I do very much agree no-useless-interpolation doesn't quite make me happy without the word "template" in it. 

Might `no-useless-template-substitution` be inoffensive to everyone, with a mention in the doc page that says 'also referred to as "string interpolation"'? That way there's no confusion what we're referring to, and people will find it with whatever search time they use.

**Comment by Josh-Cena** [comment]
"Interpolation" is the _effect_ or _purpose_ of template literals, but the code entity itself we call it an "embedded expression".

**Comment by bradzacher** [comment]
`no-useless-interpolation-expression`

**Comment by JoshuaKGoldberg** [comment]
Hmm. Accurate and precise, but "interpolation" isn't a term many devs associate with template literals... I think I'm leaning towards `no-useless-template-expression` as @Josh-Cena suggested in https://github.com/typescript-eslint/typescript-eslint/issues/8544#issuecomment-1962746645. 🤔

**Comment by kirkwaiblinger** [comment]
Hello! Just checking whether this issue has enough common ground to be actionable with a rename to `no-useless-template-expression` and doc updates? 
@JoshuaKGoldberg @bradzacher @Josh-Cena

**Comment by JoshuaKGoldberg** [comment]
Thanks for the ping - I'd say let's go with `no-useless-template-expression` for now. If someone realizes they have a strong preference later, it shouldn't be too difficult to suggest -> apply changes in the PR.
